server/channels/api4/channel.go

```diff
@@ -391,6 +391,15 @@ func patchChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// If the channel was not previously group constrained but is now, delete members that aren't part of the channel's groups
+	if patch.GroupConstrained != nil && *patch.GroupConstrained && (originalOldChannel.GroupConstrained == nil || !*originalOldChannel.GroupConstrained) {
```

Comment:
Super nit: can you rearrange the if statements to reflect the order of the comment? (or vise versa)